source/ptreferential/ptreferential_ng.cpp

```diff
@@ -101,51 +109,40 @@ std::ostream& operator<<(std::ostream& os, const BinaryOp<Diff>& op) {
     return os << '(' << op.lhs << " - " << op.rhs << ')';
 }
 
-}
+}  // namespace ast
 
 namespace {
 
 template <typename Iterator>
-struct PtRefGrammar: qi::grammar<Iterator, ast::Expr(), ascii::space_type> {
-    PtRefGrammar(): PtRefGrammar::base_type(expr) {
+struct PtRefGrammar : qi::grammar<Iterator, ast::Expr(), ascii::space_type> {
+    PtRefGrammar() : PtRefGrammar::base_type(expr) {
         using namespace qi::labels;
         using boost::spirit::_1;
         using boost::spirit::_2;
         namespace phx = boost::phoenix;
 
-        pred = all | empty | fun | cmp | dwithin;
-        all = qi::lit("all")[_val = phx::construct<ast::All>()];
+        pred  = all | empty | fun | cmp | dwithin;
+        all   = qi::lit("all")[_val = phx::construct<ast::All>()];
         empty = qi::lit("empty")[_val = phx::construct<ast::Empty>()];
-        fun = ident >> '.' >> ident >> '(' >> -(str % ',') >> ')';
-        cmp = ident[phx::at_c<0>(_val) = _1]
-            >> '.'
-            >> ident[phx::at_c<1>(_val) = _1]
-            >> '='
-            >> str[phx::push_back(phx::at_c<2>(_val), _1)];
-        dwithin = ident[phx::at_c<0>(_val) = _1]
-            >> '.'
-            >> "coord"
-            >> qi::lit("DWITHIN")[phx::at_c<1>(_val) = "within"]
-            >> '('
-            >> str[phx::push_back(phx::at_c<2>(_val), phx::construct<std::string>()),
-                   phx::push_back(phx::at_c<2>(_val), _1)]
-            >> ','
-            >> str[phx::back(phx::at_c<2>(_val)) += ";",
-                   phx::back(phx::at_c<2>(_val)) += _1]
-            >> ','
-            >> str[phx::at(phx::at_c<2>(_val), 0) = _1]
-            >> ')';
+        fun   = ident >> '.' >> ident >> '(' >> -(str % ',') >> ')';
```

Comment:
We might want to not format this, it isn't easy to understand from the start, but with everything pack it's harder

Reply:
I've surrounded the code with `// clang-format off/on`